src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -23,19 +22,21 @@ public class InMemoryExporter<T> : BaseExporter<T>
         where T : class
     {
         private readonly ICollection<T> exportedItems;
-        private readonly Func<Batch<T>, ExportResult> onExport;
+        private readonly ExportFunc onExport;
 
         public InMemoryExporter(ICollection<T> exportedItems)
         {
             this.exportedItems = exportedItems;
-            this.onExport = (Batch<T> batch) => this.DefaultExport(batch);
+            this.onExport = (in Batch<T> batch) => this.DefaultExport(in batch);
         }
 
-        internal InMemoryExporter(Func<Batch<T>, ExportResult> exportFunc)
+        internal InMemoryExporter(ExportFunc exportFunc)
         {
             this.onExport = exportFunc;
         }
 
+        internal delegate ExportResult ExportFunc(in Batch<T> batch);
```

Comment:
The `Func` delegate we were using is defined as...

```csharp
public delegate TResult Func<in T, out TResult>(T arg);
```

At a glance it looks correct, but the `in` and `out` in that signature are describing covariance/contravariance of the types, not controlling how the data is passed.

We would need...

```csharp
public delegate TResult Func<in T, out TResult>(in T arg);
```